File: day_7/7.js
/** @param {string} packages with parentheses
 *  @returns {string} Fixed and sorted packages
 */
function fixPackages(packages) {
  let result = packages;
  while (result.includes("(")) {
    const start = result.lastIndexOf("(");
    const end = result.indexOf(")", start);
    const content = result.slice(start + 1, end);
    result =
      result.slice(0, start) +
      content.split("").reverse().join("") +
      result.slice(end + 1);
  }
  return result;
}

console.log(fixPackages("a(cb)de") === "abcde");
console.log(fixPackages("a(bc(def)g)h") === "agdefcbh");
console.log(fixPackages("abc(def(gh)i)jk") === "abcighfedjk");
console.log(fixPackages("a(b(c))e") === "acbe");

File: day_7/7.py
def fix_packages(packages):
    result = packages
    while "(" in result:
        start = result.rfind("(")
        end = result.find(")", start)
        content = result[start + 1 : end]
        result = result[:start] + content[::-1] + result[end + 1 :]
    return result


print(fix_packages("a(cb)de") == "abcde")
print(fix_packages("a(bc(def)g)h") == "agdefcbh")
print(fix_packages("abc(def(gh)i)jk") == "abcighfedjk")
print(fix_packages("a(b(c))e") == "acbe")
